### PR TITLE
Provides Ask and Local instances backed by FiberRef.

### LINKED ISSUE #664: Ask and Local instances for FiberRef
So we can use Ask/Local to propagate context in a `FiberRef`. Similar to what was done [here](https://github.com/typelevel/otel4s/blob/141d739f8c8e3653c886915a76bf4e3b75ab1ddc/java/common/src/main/scala/org/typelevel/otel4s/java/instances.scala#L26) for `IOLocal`.

### PR DESCRIPTION
**Pull request by lacarvalho91**
Closes #664

### CODE REVIEW

File: zio-interop-cats-tests/shared/src/test/scala/zio/interop/CatsMtlSpec.scala [hunk 1]
```diff
@@ -21,6 +21,20 @@ class CatsMtlSpec extends ZioSpecBase {
     implicit tc => LocalTests[ZIO[Ctx, Error, _], ZEnvironment[Ctx]].local[ZEnvironment[Ctx], Int]
   )
 
+  Unsafe.unsafe { implicit unsafe =>
+    implicit val f: FiberRef[Ctx] = FiberRef.unsafe.make("")
```
Comment (lacarvalho91) [inline zio-interop-cats-tests/shared/src/test/scala/zio/interop/CatsMtlSpec.scala:25]: i wasn't sure how best to make a fiber ref in this test suite, any guidance would be appreciated

**Comment by adamgfraser** [comment]
@lacarvalho91 Looks like this needs a format!

**Comment by lacarvalho91** [comment]
@adamgfraser woops, hopefully sorted now

**Comment by adamgfraser** [comment]
@lacarvalho91 Looks like not quite!

**Comment by lacarvalho91** [comment]
@adamgfraser looks like all the checks have passed, dunno if i'm missing something :sweat_smile:

**Review by adamgfraser — APPROVED** [review]

**Pull request merged**